app/configurator/components/configurator.tsx

```diff
@@ -1,12 +1,279 @@
-import { Box } from "theme-ui";
+import React, { useRef, useState } from "react";
+import { Box, Link, LinkProps } from "theme-ui";
+import { useDebounce } from "use-debounce";
 import { useConfiguratorState } from "..";
+import { ChartPanel } from "../../components/chart-panel";
+import { ChartPreview } from "../../components/chart-preview";
+import { DataSetHint } from "../../components/hint";
+import {
+  DataCubeTheme,
+  DataCubeResultOrder,
+  useDataCubesQuery,
+} from "../../graphql/query-hooks";
+import { useLocale } from "../../src";
+import { ChartConfiguratorTable } from "../table/table-chart-configurator";
 import { ChartAnnotationsSelector } from "./chart-annotations-selector";
+import { ChartAnnotator } from "./chart-annotator";
+import { ChartConfigurator } from "./chart-configurator";
 import { ChartOptionsSelector } from "./chart-options-selector";
+import { ChartTypeSelector } from "./chart-type-selector";
 import { DataSetMetadata } from "./dataset-metadata";
-import { PanelLeft } from "./panel-left";
-import { PanelMiddle } from "./panel-middle";
+import { DataSetPreview } from "./dataset-preview";
+import {
+  SearchDatasetBox,
+  SearchFilters,
+  DatasetResults,
+} from "./dataset-search";
+import {
+  PanelLeftWrapper,
+  PanelMiddleWrapper,
+  PanelRightWrapper,
+} from "./layout";
 import { Stepper } from "./stepper";
 
+const useSearchQueryState = () => {
+  const [query, setQuery] = useState<string>("");
+
+  const [order, setOrder] = useState<DataCubeResultOrder>(
+    DataCubeResultOrder.TitleAsc
+  );
+  const previousOrderRef = useRef<DataCubeResultOrder>(
+    DataCubeResultOrder.TitleAsc
+  );
+  const [categoryFilters, setCategoryFilters] = useState<DataCubeTheme[]>([]);
+  const [includeDrafts, setIncludeDrafts] = useState<boolean>(false);
+
+  return {
```

Comment:
Nit: I guess this whole object or at least the callbacks could be wrapped in `useMemo` or `useCallback` respectively.

Reply:
Thanks, I am doing that in the step 2 PR :+1: